File: jscomp/test/gpr_3142_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = Mt.eq_suites ~test_id ~suites loc x y 


type t = 
  [ `a [@bs.as "x"] 
  | `u [@bs.as "hi"]
  | `b [@bs.as {j|你|j} ]
  | `c [@bs.as {js|我|js}]
  ]
  [@@bs.deriving jsConverter]

let v,u = tToJs, tFromJs  


(* not applicable to thiis type, and unused warning*)
#if 0 then
type t0 =  
  [ `a of int [@bs.as "hi"] ]
  [@@bs.deriving jsConverter]
#end  


;; eq __LOC__ (v `a) "x"
;; eq __LOC__ (v `u) "hi"
;; eq __LOC__ (v `b) {j|你|j}
;; eq __LOC__ (v `c) {js|我|js}

;; eq __LOC__ (u "x")  (Some `a)
;; eq __LOC__ (u "hi") (Some `u)
;; eq __LOC__ (u {j|你|j}) (Some `b)
;; eq __LOC__ (u {js|我|js}) (Some `c)
;; eq __LOC__ (u "xx") None

let () = Mt.from_pair_suites __MODULE__ !suites

File: jscomp/test/gpr_3142_test.js
'use strict';

var Mt = require("./mt.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  return Mt.eq_suites(test_id, suites, loc, x, y);
}

var _map = {"a":"x","u":"hi","b":"你","c":"我"};

var _revMap = {"x":"a","hi":"u","你":"b","我":"c"};

function tToJs(param) {
  return _map[param];
}

function tFromJs(param) {
  return _revMap[param];
}

eq("File \"gpr_3142_test.ml\", line 25, characters 6-13", tToJs("a"), "x");

eq("File \"gpr_3142_test.ml\", line 26, characters 6-13", tToJs("u"), "hi");

eq("File \"gpr_3142_test.ml\", line 27, characters 6-13", tToJs("b"), "你");

eq("File \"gpr_3142_test.ml\", line 28, characters 6-13", tToJs("c"), "我");

eq("File \"gpr_3142_test.ml\", line 30, characters 6-13", tFromJs("x"), "a");

eq("File \"gpr_3142_test.ml\", line 31, characters 6-13", tFromJs("hi"), "u");

eq("File \"gpr_3142_test.ml\", line 32, characters 6-13", tFromJs("你"), "b");

eq("File \"gpr_3142_test.ml\", line 33, characters 6-13", tFromJs("我"), "c");

eq("File \"gpr_3142_test.ml\", line 34, characters 6-13", tFromJs("xx"), undefined);

Mt.from_pair_suites("Gpr_3142_test", suites.contents);

var v = tToJs;

var u = tFromJs;

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.tToJs = tToJs;
exports.tFromJs = tFromJs;
exports.v = v;
exports.u = u;
/*  Not a pure module */
